### ISSUE TITLE
in guides.rubyonrails documentation of threads: missing info about db access

### ISSUE THREAD

**Issue by DcsMarionDickten** [rails/rails#52079]
### Steps to reproduce
I wrote a thread as documented on https://guides.rubyonrails.org/threading_and_code_execution.html.
Then  I tried to access a model dependency.

```ruby
    recalculate_archived_emails = Thread.new(company) do |c|
      execution_context = Rails.application.executor.run!
        stores = c.active_stores
        puts "all stores:"
        stores.each do |s|
          puts "#{s.name}" # so far it works
          all_emails = s.emails # hangs without error
          # lengthy computation doing something with the data

    ... 
    ensure
      execution_context.complete! if execution_context
    end
    recalculate_archived_emails.join
  end
```

### Expected behavior
Since the thread can access the company's stores it should be able to access the emails too (they are a has_many dependency).

Instead, the thread hangs without any error, blocking the whole app.

### System configuration
Rails 5.2, Mongoid, no ActiveRecord

Ruby 2.7.6

### My request
Basically I'd really appreciate some sentences (or a link) in the docs about accessing databases. A StackOverflow article told me I'd need to wrap my thread in `ActiveRecord::Base.with_connection{....}`. Maybe this should go in the guide? And also something about how I could do it without ActiveRecord?

**Comment by rafaelfranca** [comment]
`with_connection` is not needed, unless of course you are checking out the connection explicitly. How other ORM works is outside of the scope of the Rails guide. For example I have no idea why mongoid would block in that operation. Active Record would just work.
